src/txn/txn_recover.c

```diff
@@ -452,6 +452,38 @@ __recovery_set_oldest_timestamp(WT_RECOVERY *r)
     return (0);
 }
 
+/*
+ * __recovery_set_ckpt_base_write_gen --
+ *     Set the base write gen as retrieved from the metadata file.
+ */
+static int
+__recovery_set_ckpt_base_write_gen(WT_RECOVERY *r)
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    char *sys_config;
+
+    sys_config = NULL;
+    session = r->session;
+    conn = S2C(session);
+
+    /* Search the metadata for checkpoint base write gen information. */
+    WT_ERR_NOTFOUND_OK(
+      __wt_metadata_search(session, WT_SYSTEM_BASE_WRITE_GEN_URI, &sys_config), false);
+    if (sys_config != NULL) {
+        WT_CLEAR(cval);
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_BASE_WRITE_GEN, &cval));
+        if (cval.len != 0)
+            conn->txn_global.ckpt_base_write_gen = (uint64_t)cval.val;
```

Comment:
Since we have only used "conn" once, we can just replace it with S2C(session). That is what we follow in general in our code. When a function gets to the point of using "S2C" more than once, we introduce the local "conn" variable. A dist script looks for this behaviour.

Reply:
Fixed.